### PR TITLE
fix: duplicate RedPacketRecord into db

### LINKED ISSUE #1913: [Bug] Duplicate RedPacketRecord in DB
# Bug Report

### System

- [ ] Windows

  - OS Version:

- [x] Mac OS X

  - OS Version:

- [ ] Linux

  - Linux Distribution:
  - OS Version:

### Platform/Browser

- [x] Chrome

  - Maskbook Version:
  - Browser Version:

- [x] Firefox

  - Maskbook Version:
  - Browser Version:

- [ ] Android

  - Maskbook Version:
  - Android Version:

- [ ] iOS
  - Maskbook Version:
  - iOS Version:

### Build Variant

- Where do you get Maskbook?
  - [x] Store
  - [ ] ZIP
  - [ ] Self-Compiled
    - Build Commit: /_Optionally attach a Commit ID, if it is from an pre-release branch head_/

### Expect Behavior

Should check if the red packet record exists in DB before adding it in to DB.

### Actual Behavior

test123

### PR DESCRIPTION
**Pull request by septs**
closes #1913

### CODE REVIEW

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/plugins/RedPacket/database.ts [hunk 1]
```diff
@@ -15,7 +15,10 @@ export async function getRedPacket(rpid: string) {
     return record ? RedPacketRecordOutDB(record) : undefined
 }
 
-export function addRedPacket(record: RedPacketRecord) {
+export async function addRedPacket(record: RedPacketRecord) {
+    if (!isNil(await RedPacketDatabase.get('red-packet', record.id))) {
```
Comment (Jack-Works) [inline packages/maskbook/src/plugins/RedPacket/database.ts:19]: Why you need isNil?

**Comment by guanbinrui** [comment]
Testing is required.

File: packages/maskbook/src/plugins/RedPacket/database.ts (hunk 1 above)
Comment (Jack-Works) [reply packages/maskbook/src/plugins/RedPacket/database.ts:19]: It's return type is `RedPacketRecord | undefined`. It's unnecessary to use lodash

File: packages/maskbook/src/database/Plugin/wrap-plugin-database.ts [hunk 2]
```diff
@@ -28,6 +28,11 @@ export function createPluginDatabase<Data extends IndexableTaggedUnion>(plugin_i
             if (!data) return undefined
             return data.value as Of<T>
         },
+        async has<T extends Type>(type: T, id: ID<T>): Promise<boolean> {
+            const t = await c('r')
+            const count = await t.store.count(key({ type, id }))
+            return count > 1
```
Comment (septs) [inline packages/maskbook/src/database/Plugin/wrap-plugin-database.ts:34]: i need add an has method in plugin database wrapper

CC @Jack-Works

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/database/Plugin/wrap-plugin-database.ts (hunk 2 above)
Comment (Jack-Works) [reply packages/maskbook/src/database/Plugin/wrap-plugin-database.ts]: LGTM

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**